src/Nethermind/Nethermind.Blockchain/Find/BlockExtensions.cs

```diff
@@ -125,6 +125,36 @@ public static class BlockExtensions
             }
         }
 
+        public static IEnumerable<(long LogIndex, long TransactionLogIndex, TxReceipt Receipt, LogEntry Entry)> FindLogsWithReceipts(
```

Comment:
do not use tuple where you can use a struct/class

Reply:
return IEnumerable<LogWithReceipt> and limit the amount of data returned

Reply:
We could add a class/struct here. But its basically helper method. So okayish.

There is no point in limiting data here as its lazy IEnumerable based on already established collection of TxReceipts.

Reply:
I meant reviewing what is necessary inside the class. Compress (maybe compressed ratio will be 100%)

Reply:
Definitely do not leave Tuple

Reply:
I changed DTO to List<FilterLog> and moved whole method from BlockExtension to LogsSubscription. Now ReceiptInsertedEvent is an argument - have exactly this data which is needed to create FilterLogs (blockheader and txReceipts)